cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -443,7 +448,20 @@ def create_cephCSIKeyring_cephFSProvisioner(self):
         '''
         command: ceph auth get-or-create client.csi-cephfs-provisioner mon 'allow r' mgr 'allow rw' osd 'allow rw tag cephfs metadata=*'
         '''
-        cmd_json = {"prefix": "auth get-or-create",
+        metadata_pool = self._arg_parser.cephfs_metadata_pool_name
+        cluster_name = self._arg_parser.cluster_name
+        cmd_json = {}
+        if self._arg_parser.restricted_auth_permission:
+            if metadata_pool == "":
+                raise ExecutionFailureException(
+                 "'cephfs_metadata_pool_name' not found, please set the '--cephfs-metadata-pool-name' flag")
+            cmd_json = {"prefix": "auth get-or-create",
+                    "entity": "client.csi-cephfs-provisioner-%s"%cluster_name,
```

Comment:
Only append `cluster_name` if it was passed via the CLI.

Reply:
@leseb how can I differentiate between the default value or if it's passed by CLI?

Just doubting for the case if the CLI value passed was similar to default value then this would be ignored.

Reply:
What's the default value? Where is it?

Reply:
This we can handle in line#163, where we set a a `default` cluster-name.
One implementation/way is that we could pass empty-string ( "") as default cluster name. Then check if the `cluster-name` is empty, then set some variable (something like `self.DEFAULT_CLUSTER_NAME_USED=True`) and then set the default `cluster_name`.

Reply:
Its "openshift-storage", set during defining the argument [common_group.add_argument("--cluster-name", default="openshift-storage", help="Ceph cluster name")](https://github.com/rook/rook/blob/54d532b623610677df66351e18a50ced00b027f8/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L162)

Reply:
@aruniiird yaa a nice idea,
I have a question, The default value is only used at [ _excluded_keys](https://github.com/rook/rook/blob/54d532b623610677df66351e18a50ced00b027f8/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L662), how if we can remove the default value, if it doesn't effect anything?

Reply:
Exclusion is only for shell/bash type output (CLUSTER_NAME key won't be printed to the bash output). But otherwise we are using the `cluster_name` in couple of places (line#459, line#486, line#517, line#631).

Reply:
>Exclusion is only for shell/bash type output (CLUSTER_NAME key won't be printed to the bash output). 
I see it's just skipping it for bash output, so I guess not imp.
>But otherwise we are using the cluster_name in couple of places (line#459, line#486, line#517, line#631).

actually at these lines I am adding it so not a problem.

Reply:
looks good @parth-gr . Thanks